### ISSUE TITLE
Luna 4.4.0 on GTK3: theme is messed up

### ISSUE THREAD

**Issue by iksws** [guari/eclipse-ui-theme#73]
=(  Don't work fine!
Any know how to fix it?
![error](https://cloud.githubusercontent.com/assets/3934656/3443830/bdbc39ca-0127-11e4-9476-b6fec5148e41.png)

**Comment by iksws** [comment]
info!
I am use global dark theme gnome ON
window  = DUST theme
GTK+ Adwaita default
icons= default
cursor =default

**Comment by guari** [comment]
Yes, it should be a 'well-known' issue of the SWT porting on GTK3.
Starting from Luna, Eclipse is run by default with the GTK3 layer on Linux, but this option currently has also more limits on appearance customization...

You have to run Eclipse in GTK2 mode, to do it you can use a script like this to launch Eclipse:

```
#!/bin/bash
export SWT_GTK3=0
./eclipse
```

This mainly fixes the issues related to GTK3 and restores the 'old' look, except for the scrollbars: those will look worse if you use the default Ubuntu theme compared to Eclipse Kepler and Juno.

**Comment by iksws** [comment]
not work...! guari!

**Comment by guari** [comment]
put the .sh file in the same folder of the Eclipse executable and then run the script, I tested it some months ago with the same configuration and it worked.. or use a terminal ;)

**Comment by iksws** [comment]
"i try... but not work..." =(
try this
Exec=env UBUNTU_MENUPROXY=0 SWT_GTK3=0 ~/Apps/eclipse/eclipse
try this
Exec=env SWT_GTK3=0 ~/Apps/eclipse/eclipse
try in terminal...
try sh script in folder
try with eclipse kepler 

this theme not work for me =( 
this theme run only in unity?
run in terminal and i see this

./eclipse.desktop: linha 5: Unity: command not found

I desist! thks for help and time Guari!

**Comment by gustavovelascoh** [comment]
Doesn't work for me neither.

**Comment by VictorLamoine** [comment]
The fix `export SWT_GTK3=0` works for me.

**Comment by maxmil** [comment]
Does not work for me either. Looks like the button backgrounds are not being changed but the labels are. Hence you can hardly see the labels.

I'm using gnome-shell and have tried turning on and off the global shell dark theme but this seems to have no effect on eclipse.

**Comment by fdefreitas** [comment]
I'm having the exact same issue too, no luck with exports and .sh files. I'm running Ubuntu Gnome 14.04 with Gnome Shell and system dark themes, Eclipse 4.4.

**Comment by iksws** [comment]
WTF in ubuntu 14.04 Gnome with 'Dark themes', I use 
export SWT_GTK3=1 
and work like a charm!!!!!  =D

my .desktop
[Desktop Entry]
Name=Eclipse
Comment=Eclipse Luna
Exec=env SWT_GTK3=1 ~/Apps/eclipse/eclipse
Icon=~/Apps/eclipse/eclipse-1.png
StartupNotify=true
Terminal=false
Type=Application

**Comment by maxmil** [comment]
Thanks iksws this also works for me: turn on the global dark theme in gnome-tweak-tool and export SWT_GTK3=1

**Comment by gustavovelascoh** [comment]
Turning ON Gnome dark theme and exporting SWT_GTK3=1 fix buttons issue for me, but messed up item highlighting

Before (with SWT_GTK3=0, Blue highlighting, buttons messed up):
![Before (Blue highlighting, buttons messed up)](https://cloud.githubusercontent.com/assets/2431850/3628564/82eceda0-0e95-11e4-86e6-03288f01f87e.png)

After (with SWT_GTK3=1, No highlight, pretty buttons):
![After (No highlight, pretty buttons)](https://cloud.githubusercontent.com/assets/2431850/3628580/a6bbaa6e-0e95-11e4-852b-fbf61e35480d.png)

**Comment by iksws** [comment]
Yes and box lines too, but all others work, dark, dark juno and Gtk+ but
I think the file gtk3.css have problemas,tweak tool not change to gtk3-dark.css
I move gtk3-dark to gtk and have more problemas, you can see this in MENU of the gnome-terminal.

if you use gtk.css the menu is white, if you move dark.css to gtk.css the  menu is black.
I think this file have effect in all gtk interfaces.

**Comment by clopez** [comment]
Another option that worked better for me is to keep GTK3, but force a dark theme.

```
export SWT_GTK3=1
export GTK_THEME=Adwaita:dark
./eclipse
```

**Comment by spehrson** [comment]
Using GTK3 fixed the buttons for me as well, but caused all sorts of other problems  
    1) Highlighting would not work in some widgets
    2) Keyboard shortcuts and file operations would randomly stop working
    3) Git history did not display the tree view

The second issue was a deal breaker, I had to restart eclipse every 5 minutes or so.  My fix was to continue to use GTK2, but force a different theme for eclipse only.  I did this using the following script:

```
export SWT_GTK3=0
export GTK2_RC_FILES=/usr/share/darktable/darktable.gtkrc
/usr/lib64/eclipse/eclipse
```

Note 1: I did a search for dark gtkrc files on my system and decided to try the one for the darktable application before downloading random themes from the internet.  To me, the darktable theme looks amazing in eclipse!  One thing that I love about it is that it reduces the font size of all the ui elements so I see more files in the project explorer and more symbols in the outline. 

Note 2: In addition to the darktable rc files, I'm using the dark theme in eclipse and global dark in the gnome tweak tool.

Note 3: SWT_GTK3 initially didn't change anything for me because my eclipse.ini file was using --launcher.GTK_version which overrides the SWT_GTK3 environment variable.  This can be fixed by either changing the gtk version in the ini file or by deleting that option and using the environment variable instead.

**Comment by napseis** [comment]
hi,

```
export SWT_GTK3=1
export GTK_THEME=Adwaita:dark
eclipse
```

doesn't fix the buttons issues on Fedora 21.

**Comment by xoned1** [comment]
For me it's working on Fedora 21. Just: 
export SWT_GTK3=1
./eclipse 

Enjoy.

**Comment by ShahinKordasti** [comment]
Anyone done this in Linux mint? Cause I tried export SWT_GTK3=1 then start Eclipse and the buttons are still wrong color.

**Comment by OscarRL** [comment]
Fixed setting export SWT_GTK3=0 for set the GTK2 mode.. Also is useful if you use tools like JBoss  for html editing because it only supports GTK2.

Then install a trully adwaita GTK2 & GTK3 dark theme like this one:
http://cbowman57.deviantart.com/art/Adwaita-Dark-3-14-rev-2-492714039

This will also fix your Chrome and other applications that use the GTK2 theme

**Comment by gustavovelascoh** [comment]
I haven't tested this issue again for a long time (almost a year), but now it works for me using this script:

```
#!/bin/bash
export SWT_GTK3=1
~/apps/eclipse/eclipse
```

**Comment by iksws** [comment]
@gustavovelascoh
Hi man!
What distribution you are using? and what version of the eclipse?
I'm using now Mint 17/Cinnamon with luna and not work :( the gtk is a headache.

**Comment by ShahinKordasti** [comment]
Running 
export SWT_GTK3=0
Works on Eclipse Mars and Linux Mint 17.1 KDE

**Comment by mvvvv** [comment]
Running on GTK3 on unity 15.04

Both are working fine

**Comment by avinashga23** [comment]
@gustavovelascoh even i am having issues with button even after SWT_GTK3=1. I am using MINT 17.2, also had same issues in MINT 17.1.

**Comment by avinashga23** [comment]
Hi @gustavovelascoh i am able to over come this issue. here are my settings in linux mint 17.2 cinnamon.

STS settings
![screenshot from 2015-07-28 09 18 38](https://cloud.githubusercontent.com/assets/4299775/8923085/b6247ce2-3509-11e5-957f-c711f2118463.png)

Cinnamon Theme Settings

![screenshot from 2015-07-28 09 19 48](https://cloud.githubusercontent.com/assets/4299775/8923094/e2a8cb06-3509-11e5-9f19-390f942a16ed.png)

**Comment by gustavovelascoh** [comment]
I'm using Gnome Ubuntu 14.04  and Eclipse Luna. On Eclipse I use dark theme and default color and font theme, and global dark theme for Gnome. Has anyone dark theme on Eclipse Mars?

**Comment by avinashga23** [comment]
I use STS 3.7 which is based on Eclipse 4.5. No issues with Dark theme

**Comment by gustavovelascoh** [comment]
I use CodeComposerStudio based on Eclipse Version: 6.1.0.00104 , but the buttons issue is present.

**Comment by avinashga23** [comment]
I think if you can change the Ubuntu theme for controls to Delorean Dark or any dartk theme then you can over come the very annoying button issue.

**Comment by ds-hwang** [comment]
`Exec=env SWT_GTK3=0 /home/dshwang/workspace/eclipses/eclipse/eclipse` save my life. Thank you.
I use Eclipse CDT Mars.2 on Ubuntu 15.10.
